src/Parser/Mage/Fire/Modules/Items/MarqueeBindingsOfTheSunKing.js

```diff
@@ -0,0 +1,151 @@
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { formatMilliseconds, formatNumber } from 'common/format';
+import getDamageBonus from 'Parser/Mage/Shared/Modules/GetDamageBonus';
+
+const BONUS_DAMAGE = 3;
+const CAST_BUFFER = 250;
+
+const debug = false;
+
+class MarqueeBindingsOfTheSunKing extends Analyzer {
+
+  static dependencies = {
+		combatants: Combatants,
+	};
+
+  damage = 0;
+  beginCastTimestamp = 0;
+  castTimestamp = 0;
+  hasBindingsBuff = false;
+  buffUsed = false;
+  isBuffed = false;
+  beginCastFound = false;
+  count = 0;
+  wastedProcs = 0;
+  usedProcs = 0;
+  totalProcs = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasWrists(ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING.id);
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+      return;
+    }
+    this.hasBindingsBuff = true;
+    this.buffUsed = false;
+    this.totalProcs += 1;
+    this.buffAppliedTimestamp = event.timestamp;
+    debug && console.log("Buff Applied @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+      return;
+    }
+    this.hasBindingsBuff = true;
+    this.buffUsed = false;
+    this.wastedProcs += 1;
+    this.totalProcs += 1;
+    debug && console.log("Buff Refreshed " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+      return;
+    }
+    this.hasBindingsBuff = false;
+    debug && console.log("Buff Removed @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    if (this.buffUsed === false) {
+      this.wastedProcs += 1;
+      debug && console.log("Buff Expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    } else {
+      this.usedProcs += 1;
+    }
+  }
+
+  on_byPlayer_begincast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id) {
+      return;
+    }
+    this.beginCastTimestamp = event.timestamp;
+    this.beginCastFound = true;
+
+    debug && console.log("Pyroblast Begin Cast @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id || this.beginCastFound === false) {
+      return;
+    }
+    this.beginCastFound = false;
+    debug && console.log("Pyroblast Casted @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    this.castTimestamp = event.timestamp;
+    const castTime = this.castTimestamp - this.beginCastTimestamp;
+    if (castTime >= CAST_BUFFER && this.hasBindingsBuff === true) {
```

Comment:
I'm not sure I understand what this is about... I'm guessing this is to discern a hardcast pyro from an instant pyro, but would an instant pyro even have a `begincast` event? If Hot Streak Pyro does indeed have a begincast, please add a brief comment here explaining this to a future dev looking at this code.

Reply:
So when I wrote this initially, Hot Streak Pyroblast did have a begincast that was the same timestamp as the cast event. But now hot streak pyro does not have a begincast event. Nonetheless, the code still worked so i didnt change it. 

Ill take another look at it and see if i can clean it up